### PR TITLE
New packager

### PR DESCRIPTION
**Pull request by pronebird**
New packager using `electron-builder`:

- All sources are precompiled into `build/` directory.
- Dropped `electron-compile` and `electron-compilers`.
- All packages can be built using single machine regardless host OS.
- On-disk source maps provide better integration in dev tools. (e.g. inline SMs are displayed as separate files). SMs are only generated during development.

It was hard to decide between `electron-builder` and `electron-forge`. I chose `electron-builder` because it has quite narrow scope, `electron-forge` is much more powerful I think.

Relevant discussion:
https://github.com/electron-userland/electron-builder/issues/1193

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,12 +3,14 @@
   "version": "0.0.0",
   "description": "electron-react-redux-boilerplate",
   "main": "init.js",
-  "author": "Jordan Schroter",
+  "author": {
+    "name": "Jordan Schroter",
+    "email": "email@author.com"
```
Comment (pronebird) [inline package.json]: @jschr add your email if you wish :) This is needed for .deb package builds for some reason.

Comment (jschr) [inline package.json]: No need! Maybe it would be best to use a fake name too? Your call!

**Comment by jschr** [comment]
This is fantastic, great work. Nice to finally have this fixed up.

Where I got stuck last time was getting electron compiler and packager to work together and realized it be a lot simpler to just precompile using the standard babel cli rather than use electron compiler. Nice to see that you basically ended up with the same strategy.

File: package.json (hunk 1 above)
Comment (pronebird) [inline package.json]: I think this will be changed anyway since it's boilerplate project. We can leave it as is.

**Comment by develar** [comment]
> It was hard to decide between electron-builder and electron-forge

Really? Compare npm download count (200k vs 2k) and list of features :)

**Comment by jschr** [comment]
@develar Having not used forge, can you elaborate on the features from forge that you would find useful to add to this repo? 

Distributable would seem like the main advantage. It also looks like you can import existing projects into forge pretty painlessly.

**Comment by develar** [comment]
@jschr I mean opposite — electron-forge lacks a lot of features that electron-builder have. So, it is pretty easy to select between electron-forge or electron-builder. Also, you just trust stats —  200k vs 2k https://npm-stat.com/charts.html?package=electron-builder&package=electron-forge

**Comment by jschr** [comment]
@develar ah I misunderstood :) 

I think forge is still relatively new thats why it's not as popular on npm. It also uses packager behind the scenes.

**Comment by MarshallOfSound** [comment]
@develar @jschr I hate having to feel like this is a fight but I have to reiterate that `electron-forge` has a **much** wider scope that just building applications as `electron-builder`'s scope is.

The stats argument is completely useless as `electron-forge` has only been in a "MVP" state for a matter of weeks (as @jschr mentioned), whereas `electron-builder` has been around for significantly longer...

And if you really want to compare a list of features @develar I think you'll find `-forge`'s a bit longer and more wide reaching...

I would appreciate it if you took `electron-forge` as less of a "competitor" and more as a tool trying to solve a completely different problem.  We want to handle the Electron dev experience from start to end.  You are only concerned with the "end" so to speak.

We are in the same organization, following the same CoC, following the same OSS principles, please don't turn this into some kind of competition....

At the moment, I attempt to remain somewhat unbias in discussions like this and raise both `-forge` and `-builder` as potential solutions while always disclaiming the fact that `-forge` is a creation of mine.  Please attempt to do the same thing and let people make the decision themselves as to what they feel is the best path for their project.

**Comment by MarshallOfSound** [comment]
And for the record 😆 

> Distributables would seem like the main missing feature

`electron-forge` can generate windows, mac and linux distributes (zip, dmg, exe, deb, rpm) just as `-builder` can 👍

**Comment by develar** [comment]
@MarshallOfSound For projects like this (boilerplate) only building is required. i.e. only tool to build and nothing more. The same as in case of https://github.com/railsware/bozon

As was stated in the https://github.com/electron-userland/electron-builder/issues/1193 — "you don't need to choose or only electron-forge, or only electron-builder", so I am totally agree with you.

"I would appreciate it if you took electron-forge as less of a "competitor" and more as a tool trying to solve a completely different problem." Exactly. `electron-forge` is not a competitor because it is not only builder tool. And electron-builder can be used in addition easily (designed for it).

But if your project already acts as a boilerplate and you are not end user, — all electron-forge features are useless because we need only to build.

> can generate windows, mac and linux distributes (zip, dmg, exe, deb, rpm) just as

electron-builder supports more targets, including pkg, real / web installers... stop :) electron-builder is not  a competitor to electron-forge and if user happy to use electron-forge to pack and need only pkg — user can just use electron-builder only to [build pkg](https://github.com/electron-userland/electron-builder#pack-only-in-a-distributable-format).

**Pull request merged**